lib/Utils/Logging.pm

```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Save logs directly on the worker for offline upload via ulogs
+# see https://progress.opensuse.org/issues/109792
+
+package Utils::Logging;
+
+use base 'Exporter';
+use Exporter;
+use strict;
+use warnings;
+use testapi 'script_output';
+use Mojo::File qw(path);
+
+our @EXPORT = qw(log2serial);
+
+sub log2serial {
+    my $data = script_output("journalctl -k | gzip | base64 -w0");
+    my $logname = "temporary-filename.log.gz";
+    my $path = path('/tmp/ulogs')->make_path;
+    path("$path/$logname")->spurt($data);    # save the logs to the ulogs directory on the worker directly
```

Comment:
The error you were getting, in the previous runs, was because the ulogs directory was not created

```suggestion
    mkdir('ulogs') if (!-d 'ulogs');
    path("ulogs/$logname")->spurt($data);    # save the logs to the ulogs directory on the worker directly
```

Reply:
yes, this is working now, thanks!

see VR: http://openqa.suse.de/tests/9252081

I checked the logs and it looks good:
cat temporary-filename.log.gz | base64 -di | zcat